src/Context/category-context.js

```diff
@@ -0,0 +1,23 @@
+import { createContext,useState,useContext,useEffect } from "react";
+import axios from "axios";
+
+
+const CategoryContext =createContext();
+const CategoryProvider =({children})=>{
+    const [categories, setCategories] = useState([]);
+  useEffect(() => {
+    (async () => {
+      try {
+        const { data } = await axios.get("/api/categories");
+        console.log(data);
+        setCategories(data.categories);
+      } catch {
+        console.error("something went wrong");
+      }
```

Comment:
Can extract this function into a smaller API call service later on by the name as loadCategories.js

Reply:
will try to do